### PR TITLE
Increase spectests.cpp coverage

### PR DESCRIPTION
**Pull request by gumb0**
- Add remaining normal cases to `smoketest.wast`
- Add cases that will be skipped to `smoketest.wast`
- Add cases that will fail to `smoketest-failures.wast`
- Run three smoke tests on CI:
  - `smoketest.wast `
  - `smoketest.wast` with `--skip-validation`
  - `smoketest-failures.wast`

This covers most of the remaining `spectests.cpp`, the lines left not covered would mostly be hit for invalid input json / json not generated with `wast2json`.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ commands:
     - run:
         name: "Run smoketest with fizzy-spectests"
         working_directory: ~/build
-        command: bin/fizzy-spectests --skip-validation ~/project/test/spectests/smoketest
+        command: bin/fizzy-spectests ~/project/test/spectests/smoketest
```
Comment (axic) [inline circle.yml:72]: Based on the coverage you may want to run it with both options.

Comment (gumb0) [reply circle.yml:72]: Perhaps even three times, one more with failures ignored

File: test/spectests/smoketest-failures/failures.wast [hunk 2]
```diff
@@ -0,0 +1,24 @@
+;; invalid
+(module (func (drop)))
+
+;; unlinkable
+(module (import "Mod-unnknown" "foo" (func)))
```
Comment (axic) [inline test/spectests/smoketest-failures/failures.wast:5]: Is this intentionally using `unnknown` and not `unknown` ?

Comment (gumb0) [reply test/spectests/smoketest-failures/failures.wast:5]: No, thanks :)

**Pull request merged**